src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/CropEdgeGeometry.cpp

```diff
@@ -0,0 +1,410 @@
+      }
+    }
+
+    if(edgeInsideBoundary || (edgeIntersectingBoundary && behavior != BoundaryIntersectionBehavior::IgnoreEdge))
+    {
+      vertexReferenced[v0] = true;
+      vertexReferenced[v1] = true;
+      edgesMask[i] = true;
+    }
+  }
+
+  // Tally up the number of vertices referenced and edges kept
+  usize totalVerticesReferenced = std::count(vertexReferenced.begin(), vertexReferenced.end(), true);
+  usize totalEdgesKept = std::count(edgesMask.begin(), edgesMask.end(), true);
+
+  // Resize to proper sizes
+  destVertices.resizeTuples({totalVerticesReferenced});
+  destVertexAttrMatrix.resizeTuples({totalVerticesReferenced});
+  destEdges.resizeTuples({totalEdgesKept});
+  destEdgesAttrMatrix.resizeTuples({totalEdgesKept});
+
+  // Create a mapping from old vertex indices to new indices
+  std::vector<int64> vertexMapping(numVertices, -1);
+
+  int64 newIndex = 0;
+  for(usize i = 0; i < numVertices; ++i)
+  {
+    if(vertexReferenced[i])
+    {
+      vertexMapping[i] = newIndex;
+      if(behavior == BoundaryIntersectionBehavior::InterpolateOutsideVertex && interpolatedValuesMap.contains(i))
+      {
+        destVertices[3 * newIndex + 0] = std::get<0>(interpolatedValuesMap[i]);
+        destVertices[3 * newIndex + 1] = std::get<1>(interpolatedValuesMap[i]);
+        destVertices[3 * newIndex + 2] = std::get<2>(interpolatedValuesMap[i]);
+      }
+      else
+      {
+        destVertices[3 * newIndex + 0] = srcVertices[3 * i + 0];
+        destVertices[3 * newIndex + 1] = srcVertices[3 * i + 1];
+        destVertices[3 * newIndex + 2] = srcVertices[3 * i + 2];
+      }
+      newIndex++;
+    }
+  }
+
+  // Crop each vertex data array in parallel
+  {
+    ParallelTaskAlgorithm taskRunner;
+    for(const auto& [dataId, oldDataObject] : srcVertexAttrMatrix)
+    {
+      if(m_ShouldCancel)
+      {
+        return {};
+      }
+
+      const auto& oldDataArray = dynamic_cast<const IDataArray&>(*oldDataObject);
+      const std::string srcName = oldDataArray.getName();
+
+      auto& newDataArray = dynamic_cast<IDataArray&>(destVertexAttrMatrix.at(srcName));
+
+      m_MessageHandler(fmt::format("Cropping Volume || Copying Vertex Array {}", srcName));
+      ExecuteParallelFunction<CropEdgeGeomArray>(oldDataArray.getDataType(), taskRunner, oldDataArray, newDataArray, srcVertexAttrMatrix, vertexReferenced, m_ShouldCancel);
+    }
+    taskRunner.wait(); // This will spill over if the number of DataArrays to process does not divide evenly by the number of threads.
+  }
+
+  // Create final edges with remapped vertex indices
+  newIndex = 0;
+  for(usize i = 0; i < numEdges; ++i)
+  {
+    if(edgesMask[i])
+    {
+      uint64 old_v0 = srcEdges[2 * i + 0];
+      uint64 old_v1 = srcEdges[2 * i + 1];
+      int64 new_v0 = vertexMapping[old_v0];
+      int64 new_v1 = vertexMapping[old_v1];
```

Comment:
```suggestion
      // get new vertices via indexing into vertex mapping with the old index from SrcEdges
      int64 new_v0 = vertexMapping[srcEdges[2 * i + 0]];
      int64 new_v1 = vertexMapping[srcEdges[2 * i + 1]];
```
no need to explicitly copy out old index since its not used elsewhere, inline and document instead